src/hotspot/share/runtime/crac.cpp

```diff
@@ -292,11 +347,16 @@ class WakeupClosure: public ThreadClosure {
   }
 };
 
-static void wakeup_threads_in_timedwait() {
+// It requires Threads_lock to be held so it is being run as a part of VM_Operation.
+static void wakeup_threads_in_timedwait_vm() {
   WakeupClosure wc;
   Threads::java_threads_do(&wc);
+}
 
-  MonitorLocker ml(PeriodicTask_lock, Mutex::_no_safepoint_check_flag);
+// Run it after VM_Operation as it holds Threads_lock which would cause:
+// Attempting to acquire lock PeriodicTask_lock/safepoint out of order with lock Threads_lock/safepoint-1 -- possible deadlock
+static void wakeup_threads_in_timedwait() {
+  MonitorLocker ml(PeriodicTask_lock, Mutex::_safepoint_check_flag);
   WatcherThread::watcher_thread()->unpark();
 }
```

Comment:
I think it will be enough to make any update (an empty commit probably would work), but it would be cleaner to just merge the target branch into your dev branch.